sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts

```diff
@@ -224,22 +224,27 @@ export function readableSpanToEnvelope(
       throw new Error(`Unsupported span kind ${span.kind}`);
   }
 
-  envelope.data.baseData = { ...data, properties, measurements };
+  envelope.data.baseData = {
+    ...data,
+    properties,
+    measurements
+  } as RequestData | RemoteDependencyData;
   envelope.tags = tags;
   envelope.time = new Date(hrTimeToMilliseconds(span.startTime)).toISOString();
-  envelope.iKey = instrumentationKey;
-  envelope.ver = 1;
+  envelope.instrumentationKey = instrumentationKey;
+  envelope.version = 1;
 
   if (span.attributes[AzNamespace] === MicrosoftEventHub) {
-    parseEventHubSpan(span, envelope);
+    parseEventHubSpan(span, envelope as Envelope);
   } else if (span.attributes[AzNamespace]) {
     switch (span.kind) {
       case SpanKind.INTERNAL:
-        envelope.data.baseData.type = `${INPROC} | ${span.attributes[AzNamespace]}`;
+        (envelope.data
+          .baseData as RemoteDependencyData).type = `${INPROC} | ${span.attributes[AzNamespace]}`;
         break;
       default: // no op
     }
   }
 
-  return envelope;
+  return envelope as Envelope;
```

Comment:
All the casts in this function seem fragile, since we're basically opting in and out of types. 

Would it be possible to assemble the envelope at the end to avoid the partial? And perhaps to type all the base data separately before assigning to the envelope? It's a little tricky, but I feel like there should be a way to assemble a syntactically valid Envelope without having to turn off typechecks.